lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c

```diff
@@ -112,53 +112,121 @@ int encode_apn_aggregate_maximum_bit_rate(
 void bit_rate_value_to_eps_qos(
     ApnAggregateMaximumBitRate* apn_ambr, uint64_t ambr_dl, uint64_t ambr_ul) {
   uint64_t ambr_dl_kbps = ambr_dl / 1000;  // ambr_dl is expected in bps
-  uint64_t ambr_ul_kbps = ambr_ul / 1000;  // ambr_ul is expected in bps
-  if (ambr_dl_kbps == 0) {
+  uint64_t ambr_ul_kbps = ambr_ul / 1000;  // ambr_ul is expected in bps 8640000
+  apn_ambr->apnambrforuplink_extended2   = 0;
+  apn_ambr->apnambrforuplink_extended    = 0;
+  apn_ambr->apnambrfordownlink_extended2 = 0;
+  apn_ambr->apnambrfordownlink_extended  = 0;
+  apn_ambr->apnambrforuplink             = 0xff;
+  apn_ambr->apnambrfordownlink           = 0xff;
+
+  if ((ambr_dl_kbps > 256000) && (ambr_dl_kbps <= 65280000)) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION2_PRESENT;
+    apn_ambr->apnambrfordownlink_extended2 = (ambr_dl_kbps) / 256000;
+    ambr_dl_kbps =
+        ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended2) * 256000;
     apn_ambr->apnambrfordownlink = 0xff;
-  } else if ((ambr_dl_kbps > 0) && (ambr_dl_kbps <= 63)) {
-    apn_ambr->apnambrfordownlink = ambr_dl_kbps;
-  } else if ((ambr_dl_kbps > 63) && (ambr_dl_kbps <= 575)) {
-    apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 64) / 8) + 64;
-  } else if ((ambr_dl_kbps > 575) && (ambr_dl_kbps <= 8640)) {
-    apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 576) / 64) + 128;
-  } else if (ambr_dl_kbps > 8640) {
-    apn_ambr->apnambrfordownlink = 0xfe;
-    apn_ambr->extensions =
+  }
+
+  if ((ambr_dl_kbps > 8640) &&
+      ((ambr_dl_kbps >= 8600) && (ambr_dl_kbps <= 16000))) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
+    apn_ambr->apnambrfordownlink_extended = (ambr_dl_kbps - 8600) / 100;
+
+    ambr_dl_kbps =
+        ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended) * 100 - 8600;
+  }
+
+  else if ((ambr_dl_kbps > 16000) && (ambr_dl_kbps <= 128000)) {
```

Comment:
apply linter here to fix formatting.

Reply:
Ran "make format_core" to format the code.